src/Altinn.Broker.Integrations/Azure/AzureStorageService.cs

```diff
@@ -0,0 +1,217 @@
+using System.Security.Cryptography;
+
+using Altinn.Broker.Core.Domain;
+using Altinn.Broker.Core.Options;
+using Altinn.Broker.Core.Services;
+
+using Azure;
+using Azure.Storage.Blobs;
+using Azure.Storage.Blobs.Models;
+using Azure.Storage.Blobs.Specialized;
+
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+
+using Polly;
+
+namespace Altinn.Broker.Integrations.Azure;
+
+public class AzureStorageService(IResourceManager resourceManager, IOptions<AzureStorageOptions> azureStorageOptions, ILogger<AzureStorageService> logger) : IBrokerStorageService
+{
+    private async Task<BlobContainerClient> GetBlobContainerClient(FileTransferEntity fileTransferEntity, ServiceOwnerEntity serviceOwnerEntity)
+    {
+        var storageProvider = serviceOwnerEntity.GetStorageProvider(fileTransferEntity.UseVirusScan);
+        var connectionString = await resourceManager.GetStorageConnectionString(storageProvider);
+        var blobServiceClient = new BlobServiceClient(connectionString, new BlobClientOptions()
+        {
+            Retry =
+            {
+                NetworkTimeout = TimeSpan.FromHours(24),
```

Comment:
This means max upload time is 24 hours?

Reply:
Yeah, but on further thought, this makes much less sense now actually. This is the network timeout for any single operation. With the block blob client we are using now we are only uploading 32MB at once. If you're spending 24h on that operation you'll probably be killed on minimum rate restriction anyway.